### PR TITLE
Preserve cert name through ssl vc migration

### PR DESCRIPTION
**Pull request by shinrich**
With the addition of PR #6566, the origin session reuse can match can specify "cert" (cert file ATS provides to origin).  However if you are using a global look, the netvc migrate was not preserving the cert file during the migration.  This PR fixes that.

We had this information correctly moved with our internal ATS7+ but it was missing in our ATS9.  We are running with
```
proxy.config.http.server_session_sharing.pool: global
proxy.config.http.server_session_sharing.match: hostonly,sni,cert
```

The ATS9 had half the number of requests per connection to origin and 2-4 times more connections.  Once we applied this patch, the numbers started to match ATS7, since the connections that use client certificates have a much higher probability of being reusable.

This has been one of the last issues that has been preventing us for deploying ATS9 more widely.

### CODE REVIEW

File: iocore/net/UnixNetVConnection.cc [hunk 1]
```diff
@@ -1394,8 +1394,10 @@ UnixNetVConnection::migrateToCurrentThread(Continuation *cont, EThread *t)
   hold_con.move(this->con);
   SSLNetVConnection *sslvc = dynamic_cast<SSLNetVConnection *>(this);
 
-  SSL *save_ssl = (sslvc) ? sslvc->ssl : nullptr;
+  SSL *save_ssl                         = (sslvc) ? sslvc->ssl : nullptr;
+  const char *hold_ssl_client_cert_name = "";
   if (save_ssl) {
+    hold_ssl_client_cert_name = sslvc->options.ssl_client_cert_name;
```
Comment (maskit) [inline iocore/net/UnixNetVConnection.cc:1400]: Why not entire `options`? Is this the only option that need to be preserved?

Comment (shinrich) [reply iocore/net/UnixNetVConnection.cc:1400]: Yes, this is the only value that is checked during the origin session match.  The sni name name is pulled from the SSL object directly.

Comment (maskit) [reply iocore/net/UnixNetVConnection.cc:1400]: Hmm, are you suggesting I should add lines like this if I needed values in other fields? What would be a problem if we copied entire `options`?

I was surprised that this function copies only a few part of information in a NetVC. I'd be confused if I printed information in a NetVC for debugging after a migration.

Comment (shinrich) [reply iocore/net/UnixNetVConnection.cc:1400]: I guess just micro performance optimization.  Will update to copy all the options.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**